### PR TITLE
Finish profile page

### PR DESCRIPTION
**Pull request by NicholasLYang**
Added a profile page icon and way to update profile picture.

### CODE REVIEW

File: src/js/modules/core/components/Header.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ const Header: React.SFC<Props> = ({ classes, user, logout, viewportWidth }) => {
         </h1>
         {user
           ? [
-              <h1>
+              <h1 key={0}>
```
Comment (themichaelyang) [inline src/js/modules/core/components/Header.tsx]: kept throwing this warning for a while so this is v much appreciated

File: src/js/modules/core/components/ProfilePage.tsx [hunk 2]
```diff
@@ -66,15 +59,19 @@ class ProfilePage extends React.Component<Props, State> {
     const { isPasswordFormVisible } = this.state;
     this.setState({ isPasswordFormVisible: !isPasswordFormVisible });
   };
+
   render() {
     let { user, classes } = this.props;
+    const defaults = {
+      displayName: "User McUserFace",
```
Comment (themichaelyang) [inline src/js/modules/core/components/ProfilePage.tsx]: ++

File: src/js/modules/core/components/ProfilePic.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    super(props);
+    this.fileUploader = React.createRef();
+    this.state = {
+      isHovering: false,
+      isClicked: false
+    };
+  }
+
+  handleClick = () => {
+    // Gotta love how I have to do this by hand. Basically
```
Comment (themichaelyang) [inline src/js/modules/core/components/ProfilePic.tsx]: how come we gotta do it this way?

File: src/js/modules/core/components/ProfilePic.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      <div
+        className={classes.ProfilePic}
+        style={{ filter: isClicked ? "brightness(75%)" : "none" }}
+        onMouseEnter={this.handleMouseEnter}
+        onMouseLeave={this.handleMouseLeave}
+      >
+        <img
+          onClick={this.handleClick}
+          src={photoURL}
+          // Yeah yeah yeah inline styles. But I wasn't about to make a new
```
Comment (themichaelyang) [inline src/js/modules/core/components/ProfilePic.tsx]: its cool

**Comment by themichaelyang** [comment]
IMO still a pretty low priority addition, but we can merge it if the conflicts aren't too bad.

File: src/js/modules/core/components/ProfilePic.tsx (hunk 3 above)
Comment (NicholasLYang) [reply src/js/modules/core/components/ProfilePic.tsx]: Basically I can't make it a button

**Comment by NicholasLYang** [comment]
Conflicts are resolved

**Pull request merged**